File: 1833-find-the-highest-altitude/find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        prev=0
        maxi=0
        for i in range(len(gain)) :
            if maxi<prev+gain[i]:
                maxi=prev+gain[i]
            prev=prev+gain[i]
        return maxi
        

File: 1833-find-the-highest-altitude/find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int prev=0;
        int maxi=0;
        for(int i=0;i<gain.size();i++){
            if(maxi<(prev+gain[i])){
                maxi=prev+gain[i];
            }
            prev=prev+gain[i];
        }
        return maxi;

        
    }
};